src/go/rpk/pkg/api/admin/admin.go

```diff
@@ -87,24 +94,51 @@ func NewHostClient(
 }
 
 func NewAdminAPI(urls []string, tlsConfig *tls.Config) (*AdminAPI, error) {
-	return newAdminAPI(urls, tlsConfig, true)
+	return newAdminAPI(urls, tlsConfig)
 }
 
-func newAdminAPI(
-	urls []string, tlsConfig *tls.Config, tryToMapBrokerIDstoHosts bool,
-) (*AdminAPI, error) {
+func newAdminAPI(urls []string, tlsConfig *tls.Config) (*AdminAPI, error) {
 	if len(urls) == 0 {
 		return nil, errors.New("at least one url is required for the admin api")
 	}
 
+	// In situations where a request can't be executed immediately (e.g. no
+	// controller leader) the admin API does not block, it returns 503.
+	// Use a retrying HTTP client to handle that gracefully.
+	client := pester.New()
```

Comment:
The library doesn't provide much, and creating an http.Client every request is not recommended (transports are meant to be reused), but since this is a small usage for this single codepath in this PR, I think this is fine -- though I expect we will follow up later during some general maintenance to remove the lib & use our own backoff.

Reply:
Yeah, I was kind of underwhelmed with pester after I'd hooked it in.  Basically started from the realization that the standard lib didn't have baked in retry policies, so looked for what seemed popular as an off the shelf option.  For a future version of this it would be nice to have retry logic that is discriminating about what kind of error it sees before deciding what backoff policy to apply.